lib/widgets/home.dart

```diff
@@ -1,97 +1,539 @@
+import 'dart:async';
+
 import 'package:flutter/material.dart';
 
 import '../generated/l10n/zulip_localizations.dart';
 import '../model/narrow.dart';
+import 'action_sheet.dart';
+import 'app.dart';
 import 'app_bar.dart';
+import 'content.dart';
+import 'icons.dart';
 import 'inbox.dart';
+import 'inset_shadow.dart';
 import 'message_list.dart';
 import 'page.dart';
+import 'profile.dart';
 import 'recent_dm_conversations.dart';
 import 'store.dart';
 import 'subscription_list.dart';
 import 'text.dart';
+import 'theme.dart';
+
+enum HomePageTab {
+  inbox,
+  channels,
+  directMessages,
+}
 
-class HomePage extends StatelessWidget {
-  const HomePage({super.key});
+class HomePage extends StatefulWidget {
+  const HomePage({super.key, required this.initialTab});
 
-  static Route<void> buildRoute({required int accountId}) {
+  static Route<void> buildRoute({required int accountId, HomePageTab? initialTab}) {
     return MaterialAccountWidgetRoute(accountId: accountId,
-        page: const HomePage());
+      loadingPlaceholderPage: const _LoadingPlaceholderPage(),
+      page: HomePage(initialTab: initialTab ?? HomePageTab.inbox));
   }
 
+  /// Navigate to [HomePage], ensuring that its route is at the root level.
+  static void navigate(BuildContext context, {required int accountId}) {
+    final navigator = Navigator.of(context);
+    navigator.popUntil((route) => route.isFirst);
+    unawaited(navigator.pushReplacement(
+      HomePage.buildRoute(accountId: accountId)));
+  }
+
+  final HomePageTab initialTab;
+
   @override
-  Widget build(BuildContext context) {
-    final store = PerAccountStoreWidget.of(context);
+  State<HomePage> createState() => _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
+  late final ValueNotifier<HomePageTab> _tab = ValueNotifier(widget.initialTab);
+
+  String get _currentTabTitle {
```

Comment:
nit: put this just above `build`, after the state management; it's effectively a private helper for `build`